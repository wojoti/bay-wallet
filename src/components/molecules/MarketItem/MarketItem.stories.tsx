import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {MarketItem} from 'components/molecules';

export default {
  title: 'Molecules/MarketItem',
  component: MarketItem,
} as ComponentMeta<typeof MarketItem>;
const Template: ComponentStory<typeof MarketItem> = args => (
  <MarketItem {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  value: '0.02137',
  name: 'Bitcoin',
  percent: '4',
};
