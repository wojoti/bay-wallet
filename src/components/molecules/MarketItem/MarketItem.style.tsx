import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  wrapper: {
    marginVertical: 13,
    paddingHorizontal: 30,
  },
  circle: {
    backgroundColor: '#346EC5',
    borderRadius: 45,
    width: 45,
    height: 45,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 15,
  },
  detailsSpacer: {width: 'auto', marginRight: 10},
});
export default style;
