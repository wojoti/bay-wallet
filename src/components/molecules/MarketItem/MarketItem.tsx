import {Container, Header, Img, ImgType, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {grey, lime, marketDarkGrey, red} from 'theme/colors';
import style from './MarketItem.style';

export type MarketItemProps = PropsWithChildren<{
  value: string;
  name: string;
  percent: string;
  testId?: string;
}>;

const MarketItem = ({value, name, percent, testId}: MarketItemProps) => {
  const detailColor = percent.startsWith('-') ? red : lime;
  const isDown = percent.startsWith('-');
  return (
    <Container
      testId={testId}
      direction="row"
      align="center"
      justify="space-between"
      customStyle={style.wrapper}>
      <Container flex={5} direction={'row'} align="center">
        <Container customStyle={style.circle}>
          <Img src={ImgType.bitcoin} height={'20'} alt={'bitcoin logo'} />
        </Container>
        <Container justify="center" align="flex-start">
          <Header color={marketDarkGrey} size={20}>
            $ {value}
          </Header>
          <Label color={grey} size={12}>
            {name}
          </Label>
        </Container>
      </Container>
      <Container flex={1} direction={'row'} justify="flex-end" align="center">
        <Container customStyle={{width: 'auto', marginRight: 10}}>
          <Header size={14} color={detailColor}>
            {percent}%
          </Header>
        </Container>
        {isDown ? (
          <Img src={ImgType.dashDown} height={'12'} width={12} alt={''} />
        ) : (
          <Img src={ImgType.dashUp} height={'12'} width={12} alt={''} />
        )}
      </Container>
    </Container>
  );
};

export default MarketItem;
