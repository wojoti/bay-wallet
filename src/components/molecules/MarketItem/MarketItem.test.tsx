import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import MarketItem, {MarketItemProps} from './MarketItem';

const props: MarketItemProps = {
  testId: 'test-marketitem-id',
  value: '1234',
  name: 'name',
  percent: '4',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<MarketItem {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render MarketItem', () => {
  render(<MarketItem {...props} />);
  const MarketItemElement = screen.getByTestId('test-marketitem-id');
  expect(MarketItemElement).toBeOnTheScreen();
});
