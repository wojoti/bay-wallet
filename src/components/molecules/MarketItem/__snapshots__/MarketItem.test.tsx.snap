// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
      {
        "marginVertical": 13,
        "paddingHorizontal": 30,
      },
    ]
  }
  testID="test-marketitem-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "alignItems": "center",
            "backgroundColor": "#346EC5",
            "borderRadius": 45,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 15,
            "width": 45,
          },
        ]
      }
    >
      <SvgMock
        height={20}
        testID="marketitem-img"
        width="100%"
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#3D454B",
            "fontFamily": "Poppins-Bold",
            "fontSize": 20,
            "letterSpacing": undefined,
            "textAlign": "left",
          }
        }
        testID="marketitem-value"
      >
        $ 
        1234
      </Text>
      <Text
        style={
          {
            "color": "#6B6B6B",
            "fontFamily": "Poppins-Light",
            "fontSize": 12,
            "fontWeight": "400",
            "textAlign": "left",
          }
        }
        testID="marketitem-name"
      >
        name
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "marginRight": 10,
            "width": "auto",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#53E520",
            "fontFamily": "Poppins-Bold",
            "fontSize": 14,
            "letterSpacing": undefined,
            "textAlign": "left",
          }
        }
        testID="marketitem-percent"
      >
        4
        %
      </Text>
    </View>
    <SvgMock
      height={12}
      testID="marketitem-dash"
      width={12}
    />
  </View>
</View>
`;
